stdlib/public/Atomics/AtomicMemoryOrderings.swift

```diff
@@ -0,0 +1,287 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Swift
+
+/// Specifies the memory ordering semantics of an atomic load operation.
+@frozen
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+public struct AtomicLoadOrdering {
+  // This struct works like a non-frozen enum whose cases aren't reorderable.
+
+  @usableFromInline
+  internal var _rawValue: Int
+
+  @_semantics("constant_evaluable")
+  @inlinable @_transparent // Debug performance
+  internal init(_rawValue: Int) {
+    self._rawValue = _rawValue
+  }
+}
+
+@available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+extension AtomicLoadOrdering {
+  // FIXME: Explain these ordering levels in more detail.
+
+  /// Guarantees the atomicity of the specific operation on which it is applied,
+  /// but imposes no ordering constraints on any other reads or writes.
+  @_semantics("constant_evaluable")
+  @_alwaysEmitIntoClient
+  @_transparent // Debug performance
+  public static var relaxed: AtomicLoadOrdering {
+    AtomicLoadOrdering(_rawValue: 0)
```

Comment:
These properties and the `Equatable.==` functions could use `Self` from [SE-0068](https://github.com/apple/swift-evolution/blob/master/proposals/0068-universal-self.md).

```suggestion
  public static var relaxed: Self {
    Self(_rawValue: 0)
```